Header in facility detail page now focuses

gescobardev opened the pull request:
## Description
Original issue linked below

When tabbing through the facility detail page AFTER you have selected one from the search results in Facility Locator, the header (h1) will now focus.

To test you can use a facility from El Paso- http://localhost:3001/find-locations/?zoomLevel=9&page=1&address=el%20paso&location=undefined%2Cundefined&context=El%20Paso%2C%20Texas%2C%20United%20States&facilityType=health&serviceType

## Acceptance criteria
- [ ] VoiceOver announces the header(h1) of the facility detail page.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

Commit:
Header is now focusable in facility detail page

Commit:
cleaned code

mpelzsherman committed:
Move header focus to componentDidUpdate

mpelzsherman commented on src/applications/facility-locator/containers/FacilityDetail.jsx:
```diff
@@ -111,7 +113,10 @@ class FacilityDetail extends Component {
     return (
       <div>
-        <h1>{name}</h1>
+        {/* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */}
```
@joshkimux - We have a lint rule that tells us to avoid making non-interactive elements focusable. Do you think we should disable that rule?

joshkimux replied:
Will using tabindex="-1" still trigger this?

mpelzsherman replied:
No, it doesn't. But we are having to disable the rule a LOT:

https://github.com/department-of-veterans-affairs/vets-website/search?q=no-noninteractive-tabindex

... which makes me concerned that we are applying this rule inconsistently on the site.

gescobardev replied:
resolving convo - more details provided in comments below

mpelzsherman committed:
Disable eslint rule jsx-a11y/no-noninteractive-tabindex

Why?
- our a11y team has requested that we focus h1 elements specifically
- we are disabling this rule in many places throughout the codebase

Rleahy22 commented on src/applications/facility-locator/containers/FacilityDetail.jsx:
```diff
@@ -111,7 +110,9 @@ class FacilityDetail extends Component {
     return (
       <div>
-        <h1>{name}</h1>
+        <h1 ref={this.headerRef} tabIndex={0}>
```
should `tabIndex` be set to -1 so we can set focus to it, but the user cannot tab to it as it is not an interactive element?

joshkimux replied:
Yes, tabIndex should be set to -1!

gescobardev replied:
Yes, thank you Ryan. I'll make that change.

zacharymorel replied:
Learn something new with this tabIndex attribute!

mpelzsherman commented:
I think the intention is that users should be able to focus on the H1. Josh, can you please clarify?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Ryan Leahy ***@***.***>
Sent: Thursday, July 29, 2021 1:27:58 PM
To: department-of-veterans-affairs/vets-website ***@***.***>
Cc: Michael Pelz-Sherman ***@***.***>; Comment ***@***.***>
Subject: Re: [department-of-veterans-affairs/vets-website] Header in facility detail page now focuses (#18050)


@Rleahy22 commented on this pull request.

________________________________

In src/applications/facility-locator/containers/FacilityDetail.jsx<https://github.com/department-of-veterans-affairs/vets-website/pull/18050#discussion_r679349622>:

> @@ -111,7 +110,9 @@ class FacilityDetail extends Component {
     const isVBA = facilityType === FacilityType.VA_BENEFITS_FACILITY;
     return (
       <div>
-        <h1>{name}</h1>
+        <h1 ref={this.headerRef} tabIndex={0}>


should tabIndex be set to -1 so we can set focus to it, but the user cannot tab to it as it is not an interactive element?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/department-of-veterans-affairs/vets-website/pull/18050#pullrequestreview-718356189>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAATTC6Y4YDQTL4ITIXSLBDT2GFR5ANCNFSM5BFFKM2Q>.

joshkimux commented:
The intent is for focus to be placed on the `h1` on page load when coming from the facilities locator search 😄 
It should _not_ be within the natural tab order, so tabindex="-1" should be used.

mpelzsherman commented:
thanks for the clarification Josh. @Gaida ***@***.***> please change as per above. Hopefully we will not need the eslint rule to be disabled. I am curious why it needs to be disabled so often in other parts of our code though.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Josh Kim ***@***.***>
Sent: Thursday, July 29, 2021 1:31:25 PM
To: department-of-veterans-affairs/vets-website ***@***.***>
Cc: Michael Pelz-Sherman ***@***.***>; Comment ***@***.***>
Subject: Re: [department-of-veterans-affairs/vets-website] Header in facility detail page now focuses (#18050)


@joshkimux commented on this pull request.

________________________________

In src/applications/facility-locator/containers/FacilityDetail.jsx<https://github.com/department-of-veterans-affairs/vets-website/pull/18050#discussion_r679352045>:

> @@ -111,7 +110,9 @@ class FacilityDetail extends Component {
     const isVBA = facilityType === FacilityType.VA_BENEFITS_FACILITY;
     return (
       <div>
-        <h1>{name}</h1>
+        <h1 ref={this.headerRef} tabIndex={0}>


Yes, tabIndex should be set to -1!

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/department-of-veterans-affairs/vets-website/pull/18050#discussion_r679352045>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAATTC3IKQRQ35JRC5TMJU3T2GF63ANCNFSM5BFFKM2Q>.

gescobardev commented:
@joshkimux ahhh ok, thanks for the clarification. I thought we did want the `h1` in the tab order. I'll make the change!

Commit:
changed h1 tabindex to -1 and removed eslintrc.js tabindex disable code
